### ISSUE TITLE
PowerShell Universal Service Startup Delay (15-30 minutes)

### ISSUE THREAD

**Issue by RubenT91** [ironmansoftware/powershell-universal#3280]
### Version

4.2.15

### Severity

Low

### Environment

msi

### Steps to Reproduce

PSU going through loading the endpoints and then no logs pertaining to loading endpoints until finally, at a much later time, it picks up without error. All of the endpoints work fine. The intervals of the pause makes me think of something timing out as it is always like nearly exactly 20 minutes like the screenshot below.

1. **Monitoring and Resource Utilization**: Monitored the system during the startup process and found minimal resource utilization, with no apparent spikes in CPU, memory, or disk I/O.

2. **Restarts and Upgrades**: Tried restarting the PowerShell Universal service and the entire server, as well as upgrading to the latest version of PowerShell Universal, but the issue persists.

3. **Configuration and Script Changes**: Any recent changes to the PowerShell Universal configuration or the underlying scripts, ruling out those potential contributors.

### Expected behavior

```console
Minimal delay on the start process
```

### Actual behavior

```console
Significant startup delay (15-30 minutes) with the PowerShell Universal service in the environment, which is impacting the availability and responsiveness of applications.

From the screenshot below, review the following:

- The script (line 7120) where we see load in the logs I provided then the wait occurred after simply outputs “Bleep Bloop”, nothing else.  
- The script (line 23081) is fairly small and it only contains parameters, executes a PSU job asynchronously, then returns the job ID.
- However, in troubleshooting I have also removed all code from scripts and it is the same thing. We only have a few API endpoints and I believe even if I removed one of the two scripts above it would still freeze on another. I don’t believe it freezes on just what is in this log but its been a few weeks since we rebooted the service last and while I had collected those logs I have since lost them.
```

### Additional Environment data

- Windows Server 2022
- SQL Server 2022
- PowerShell Universal version 4.2.15

## Hardware Specifications

- 16 Core CPU
- 64GB Memory
- 100GB Enterprise Flash Array Storage

### Screenshots/Animations

attached to ticket

**Comment by adamdriscoll** [comment]
@RubenT91 - Can you have the customer capture a memory dump while this is happening?

**Comment by RockoTheeHut** [comment]
Found the culprit. We have two environments, loads the same module, one has persistence turned on, the other does not. One of them had a credential defined for the environment. That environment is what caused the endpoints to freeze the loading for long periods of time. Using the same environment in scripts does not cause the issue. The other environment, without the credential, doesn't cause issues. 

We didn't need the credential defined so I was able to remove it and endpoints load relatively quickly now. Feels like a bug of some sort.. But for now we have a good solution to the loading times.
